CMakeLists.txt

```
@@ -74,14 +75,32 @@ if(MSVC)
   # Produce object files that contain debug info.
   target_compile_options(valkey PRIVATE /Z7)
 endif()
+
+set(valkey_link_libraries)
+set(valkey_compile_definitions)
+
 IF(WIN32)
-    TARGET_LINK_LIBRARIES(valkey PUBLIC ws2_32 crypt32)
+    list(APPEND valkey_link_libraries ws2_32 crypt32)
 ELSEIF(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
-    TARGET_LINK_LIBRARIES(valkey PUBLIC m)
+    list(APPEND valkey_link_libraries m)
 ELSEIF(CMAKE_SYSTEM_NAME MATCHES "SunOS")
-    TARGET_LINK_LIBRARIES(valkey PUBLIC socket)
+    list(APPEND valkey_link_libraries socket)
 ENDIF()
 
+IF(ENABLE_THREADS)
+    FIND_PACKAGE(Threads REQUIRED)
```

We can use lowercase on all touched CMake commands, which is recommended style these days.
```suggestion
    find_package(Threads REQUIRED)
```
same for `if`, `endif`, `TARGET_LINK_LIBRARIES`